homeassistant/components/recorder/__init__.py

```diff
@@ -864,20 +930,37 @@ def async_external_statistics(self, metadata: dict, stats: Iterable[dict]) -> No
         """Schedule external statistics."""
         self.queue.put(ExternalStatisticsTask(metadata, stats))
 
+    @callback
+    def _using_sqlite(self) -> bool:
+        return bool(self.engine and self.engine.dialect.name == "sqlite")
+
     @callback
     def _async_setup_periodic_tasks(self) -> None:
         """Prepare periodic tasks."""
         if self.hass.is_stopping or not self.get_session:
             # Home Assistant is shutting down
             return
 
+        # If the db is using a socket connection, we need to keep alive
+        # to prevent errors from unexpected disconnects
+        if not self._using_sqlite():
+            self._keep_alive_listener = async_track_time_interval(
+                self.hass, self._async_keep_alive, timedelta(seconds=KEEPALIVE_TIME)
+            )
+
+        # If the commit interval is not 0, we need commit periodicly
```

Comment:
we need to commit periodically

Reply:
https://github.com/home-assistant/core/pull/69658